### DISCUSSION TITLE
Multiple outputs per step - adaptivity

### DISCUSSION THREAD

**Discussion by HPL001** [idaholab/moose#29536] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

I used the following output Settings, but I didn't get the output for each time step. Instead, I get the output for each iteration of each time step. How should I set this up？
```
[Outputs]
  exodus = true
  print_linear_residuals = true
  execute_on = timestep_end
  [out]
    type = VTK # 输出 VTK 文件
    time_step_interval = 1 # 每个时间步输出
    file_base = Sintering_output # 输出文件的基础名称
    enable = true
    execute_on = 'timestep_end'
  []
```
![image](https://github.com/user-attachments/assets/e4083ed2-d775-447b-9ada-a1bdb9a7585e)

**Comment by GiudGiud** [comment]
hello

remove this it's not needed
```
    time_step_interval = 1 # 每个时间步输出
```
Where are the iterations coming from? From the nonlinear solver or from fixed point iterations with a MultiApp?

**Reply by HPL001** [reply]
Thank you for your reply, specifically, my execution part is this
```
[Executioner]
  type = Transient
  scheme = BDF2
  solve_type = PJFNK
  petsc_options_iname = '-pc_type -ksp_gmres_restart -sub_ksp_type
                        -sub_pc_type -pc_asm_overla'
  petsc_options_value = 'asm      31       preonly
                         ilu      2'
  l_tol = 1e-05
  nl_max_its = 30
  l_max_its = 50
  nl_rel_tol = 1e-07
  nl_abs_tol = 1e-09
  start_time = 0.0
  end_time = 1.0
  # dt = 0.00125

  [TimeStepper]
    type = IterationAdaptiveDT
    cutback_factor = 0.5
    dt = 0.001
    growth_factor = 1.1
    optimal_iterations = 15
  []
[]
```

**Reply by GiudGiud** [reply]
i think these multiple files per time step are from steps of adaptivity. 
You can try to output on timestep begin instead of timestep_end to (maybe) prevent outputting multiple files per time step

**Reply by HPL001** [reply]
Thanks for your reply, I have modified the input file according to your suggestion, but it seems to have no effect.
```
[Outputs]
  exodus = true
  print_linear_residuals = true
  execute_on = timestep_end
  [out]
    type = VTK
    file_base = Sintering_output
    enable = true
    execute_on = 'timestep_begin'
  []
  # [out]
  #   type = Tecplot
  #   ascii_append = true
  # []

  [debug] # report variable residual
    type = VariableResidualNormsDebugOutput
  []
[]
```

**Reply by HPL001** [reply]
@GiudGiud Hello dear developer, can you continue to give me some advice.

**Reply by GiudGiud** [reply]
ok then we dont have an option to do this. You would need to modify the FEProblemSolve.C class to output at the end of the adaptivity loop, if I am correct that these files are from adaptivity. You can open them all for a time step and check that the only difference is the local mesh refinement. 

Have too many output files is not really preventing you from doing anything though so I would recommend you move on

**Reply by HPL001** [reply]
Thank you for your reply, which is a good solution to my problem
